### PR TITLE
Add support for removing nulls with array_remove

### PR DESCRIPTION
**Pull request by nezihyigitbasi**
Postgres already supports removing `null`s from arrays that contain `null` values, this PR adds the same functionality to the `array_remove` function.

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/type/TestArrayOperators.java [hunk 1]
```diff
@@ -690,6 +690,11 @@ public void testArrayRemove()
         assertFunction("ARRAY_REMOVE(ARRAY [TRUE, FALSE, TRUE], FALSE)", new ArrayType(BOOLEAN), ImmutableList.of(true, true));
         assertFunction("ARRAY_REMOVE(ARRAY [NULL, FALSE, TRUE], TRUE)", new ArrayType(BOOLEAN), asList(null, false));
         assertFunction("ARRAY_REMOVE(ARRAY [ARRAY ['foo'], ARRAY ['bar'], ARRAY ['baz']], ARRAY ['bar'])", new ArrayType(new ArrayType(VARCHAR)), ImmutableList.of(ImmutableList.of("foo"), ImmutableList.of("baz")));
+        assertFunction("ARRAY_REMOVE(ARRAY [1, 2, NULL, 3, NULL, 4, 5], NULL)", new ArrayType(BIGINT), asList(1L, 2L, 3L, 4L, 5L));
+        assertFunction("ARRAY_REMOVE(ARRAY ['foo', NULL, 'bar', NULL], NULL)", new ArrayType(VARCHAR), asList("foo", "bar"));
+        assertFunction("ARRAY_REMOVE(ARRAY [1.0, NULL, nan(), -1.23, NULL], NULL)", new ArrayType(DOUBLE), asList(1.0, NaN, -1.23));
+        assertFunction("ARRAY_REMOVE(ARRAY [TRUE, NULL, FALSE, TRUE, NULL], NULL)", new ArrayType(BOOLEAN), asList(true, false, true));
+//        assertFunction("ARRAY_REMOVE(ARRAY [NULL, NULL, NULL], NULL)", new ArrayType(UNKNOWN), asList());
```
Comment (nezihyigitbasi) [inline presto-main/src/test/java/com/facebook/presto/type/TestArrayOperators.java]: This somehow works through the cli, but fails in unit tests

```
presto:nyigitbasi> select ARRAY_REMOVE(ARRAY [NULL, NULL, NULL], NULL);
 _col0
-------
 []
(1 row)
```

**Comment by cberner** [comment]
@martint what do you think about allowing `array_remove` to remove `NULL`? We haven't really specified how `NULL` in an array works

**Comment by martint** [comment]
I think that'd be weird given that NULL in SQL means "unknown value". I'd rather wait for when we have lambdas and then you can do `filter(array, x -> x is null)`

**Comment by nezihyigitbasi** [comment]
Thanks guys, when do you think lambdas will be available?

**Comment by cberner** [comment]
I'm hoping next quarter

**Pull request closed without merging**